packages/chord-detect/index.ts

```diff
@@ -19,21 +19,48 @@ const namedSet = (notes: string[]) => {
   return (chroma: number) => pcToName[chroma];
 };
 
-export function detect(source: string[]): string[] {
+export function detect(source: string[], assumePerfectFifth: boolean = false): string[] {
   const notes = source.map((n) => note(n).pc).filter((x) => x);
   if (note.length === 0) {
     return [];
   }
 
-  const found: FoundChord[] = findExactMatches(notes, 1);
+  const found: FoundChord[] = findExactMatches(notes, 1, assumePerfectFifth);
 
   return found
     .filter((chord) => chord.weight)
     .sort((a, b) => b.weight - a.weight)
     .map((chord) => chord.name);
 }
 
-function findExactMatches(notes: string[], weight: number): FoundChord[] {
+// 3m 000100000000
+// 3M 000010000000
+// 5d 000000100000
+// 5P 000000010000
+// 5A 000000001000
+// 7m 000000000010
+// 7M 000000000001
+function has3x_5P_7x(chordType: ChordType) {
+  const cromaNum = parseInt(chordType.chroma, 2)
+  return ((cromaNum & 384) && (cromaNum & 16) && (cromaNum & 3))
```

Comment:
I like the bitwise operator approach but makes code really difficult to understand.

A possible solution could be creating helper functions:
```ts
// 3m 000100000000
// 3M 000010000000
const hasAnyThird = (chroma) => !!(cromaNum & 384)

// 5d 000000100000
const hasPerfectFifth = ...
...

Or alternatively:

```ts
// 3m 000100000000
// 3M 000010000000
const hasAnyThird = 384

const  testChroma = (chroma: number, tester: number) => !!(chroma & tester)
```

and then

```ts
return hasAnyThird(chroma) && has...

// or

return testChroma(chroma, hasAnyThird) && ..
```